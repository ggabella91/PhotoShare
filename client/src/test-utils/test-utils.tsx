import { FC, ReactElement } from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render, RenderOptions } from '@testing-library/react';
import { store } from '../redux/store';

interface WrapperProps {
  wrapperProps?: { route: string; location: string };
}

type CustomRenderOptions = RenderOptions & WrapperProps;

const WithReduxProvider: FC<WrapperProps> = ({ children, wrapperProps }) => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        <Routes>
          <Route path={wrapperProps?.route || '/'} element={children} />
        </Routes>
      </BrowserRouter>
    </Provider>
  );
};

const renderWithReduxAndRouter = (
  ui: ReactElement,
  options?: Omit<CustomRenderOptions, 'wrapper'>
) => {
  window.history.pushState({}, '', options?.wrapperProps?.location || '/');
  const customContainer = document.createElement('div');
  customContainer.id = 'root';

  return render(ui, {
    wrapper: (props) => (
      <WithReduxProvider {...props} wrapperProps={options?.wrapperProps} />
    ),
    container: document.body.appendChild(customContainer),
    ...options,
  });
};

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export { renderWithReduxAndRouter as render };
