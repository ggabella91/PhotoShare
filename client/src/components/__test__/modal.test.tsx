import { render } from '../../test-utils/test-utils';
import CustomModal from '../modal/modal.component';

it('renders a confirmation modal component', () => {
  const { container: customModal } = render(
    <CustomModal
      header='header-text'
      subheader='subheader-text'
      bodytext='body-text'
      show={true}
      onHide={() => {}}
      handleconfirm={() => {}}
      actionlabel='action-label'
    />
  );

  expect(customModal).toBeInTheDocument();
});
