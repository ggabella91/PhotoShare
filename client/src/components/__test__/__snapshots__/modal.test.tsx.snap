// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a confirmation modal component 1`] = `
<Modal
  actionlabel="action-label"
  animation={true}
  autoFocus={true}
  backdrop={true}
  bodytext="body-text"
  centered={true}
  dialogAs={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  dialogClassName="custom-modal"
  enforceFocus={true}
  handleconfirm={[Function]}
  header="header-text"
  keyboard={true}
  onHide={[Function]}
  restoreFocus={true}
  show={true}
  subheader="subheader-text"
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      id="contained-modal-title-vcenter"
    >
      header-text
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    className="body-text"
  >
    <h4>
      subheader-text
    </h4>
    <p>
      body-text
    </p>
  </ModalBody>
  <ModalFooter>
    <Button
      className="button modal-button"
      onClick={[Function]}
    >
      Cancel
    </Button>
    <Button
      className="button modal-button delete-button"
      onClick={[Function]}
    >
      action-label
    </Button>
  </ModalFooter>
</Modal>
`;
